core/src/main/scala/org/bitcoins/core/crypto/ExtKey.scala

```diff
@@ -42,16 +44,38 @@ sealed abstract class ExtKey extends NetworkElement {
   /** The key at this path */
   def key: BaseECKey
 
+  /**
+    * Derives the child pubkey at the specified index
+    */
   def deriveChildPubKey(idx: UInt32): Try[ExtPublicKey] = this match {
     case priv: ExtPrivateKey =>
       Success(priv.deriveChildPrivKey(idx).extPublicKey)
     case pub: ExtPublicKey => pub.deriveChildPubKey(idx)
   }
 
+  /**
+    * Derives the child pubkey at the specified index
+    */
   def deriveChildPubKey(idx: Long): Try[ExtPublicKey] = {
     Try(UInt32(idx)).flatMap(deriveChildPubKey)
   }
 
+  /**
+    * Derives the child pubkey at the specified path
+    */
+  def deriveChildPubKey(path: BIP32Path): Try[ExtPublicKey] = {
+    this match {
+      case priv: ExtPrivateKey =>
+        Success(priv.deriveChildPrivKey(path).extPublicKey)
+      case pub: ExtPublicKey =>
+        Try {
```

Comment:
An alternative that doesn't throw and preserves the original error message, but is longer

```scala
        @tailrec
        def loop(
            remainingPath: List[BIP32Child],
            accum: ExtPublicKey): Try[ExtPublicKey] = {
          remainingPath match {
            case h :: t =>
              val derivedPubT = accum.deriveChildPubKey(h.toUInt32)
              if (derivedPubT.isSuccess) {
                loop(t, derivedPubT.get)
              } else {
                derivedPubT
              }
            case Nil => Success(accum)
          }
        }
        loop(path.children.toList, pub)
```